chronicles.nim

```diff
@@ -255,6 +262,15 @@ macro logIMPL(lineInfo: static InstInfo,
   code.add newCall("logAllDynamicProperties", Stream, record)
   code.add newCall("flushRecord", record)
 
+  when compileOption("threads"):
```

Comment:
Sorry, I've changed my mind. We don't need locking here at all. Instead, we can unconditionally use the `BufferedOutput` type when compiling with `threads: on`. Then, the `setProperty` statements will populate a local buffer that will be flushed in the end with a single operation. Only, the actual output device needs to be thread-safe, which is already the case for `printf`.

The above can be achieved by modifying the function below:
https://github.com/status-im/nim-chronicles/blob/master/chronicles/log_output.nim#L155